geohash_test.go

```diff
@@ -101,3 +101,66 @@ func TestEncode(t *testing.T) {
 		}
 	}
 }
+
+type adjacentTest struct {
+	geohash string
+	dir string
+	adjacent string
+}
+
+func TestCalculateAdjacent(t *testing.T) {
+	var tests = []adjacentTest{
+		adjacentTest {
+			geohash: "zbzury",
+			dir: "right",
+			adjacent: "b0bh2n",
+		},
+		adjacentTest {
+			geohash: "b0bh2n",
+			dir: "bottom",
+			adjacent: "b0bh2j",
+		},
+		adjacentTest {
+			geohash: "b0bh2j",
+			dir: "left",
+			adjacent: "zbzurv",
+		},
+		adjacentTest {
+			geohash: "zbzurv",
+			dir: "left",
+			adjacent: "zbzurt",
+		},
+		adjacentTest {
+			geohash: "zbzurt",
+			dir: "up",
+			adjacent: "zbzurw",
+		},
+		adjacentTest {
+			geohash: "zbzurw",
+			dir: "up",
+			adjacent: "zbzurx",
+		},
+		adjacentTest {
+			geohash: "zbzurx",
+			dir: "right",
+			adjacent: "zbzurz",
+		},
+		adjacentTest {
+			geohash: "zbzurz",
+			dir: "right",
+			adjacent: "b0bh2p",
+		},
+		adjacentTest {
+			geohash: "b0bh2p",
+			dir: "bottom",
+			adjacent: "b0bh2n",
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
if you encapsulate each test in a `t.Run(fmt.Sprintf("%s -> %s  -> %s", test.geohash, test.dir, test.adjacent), func(t *testing.T) { ... })` then you'll get a nicer output and you'll be able to filter testcases

Reply:
Thanks, the thing is I've just wanted to follow the existing structure of the file.